envoy_reader/envoy_reader.py

```diff
@@ -133,8 +133,6 @@ async def getData(self):  # pylint: disable=invalid-name
                         response.raise_for_status()
                 except (httpcore.RemoteProtocolError, httpx.RemoteProtocolError):
                     continue
-                except httpx.HTTPError:
-                    response.raise_for_status()
```

Comment:
Fixes
```
2021-04-08 02:04:54 ERROR (MainThread) [homeassistant.components.enphase_envoy] Unexpected error fetching envoy {name} data: local variable 'response' referenced before assignment
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/httpx/_exceptions.py", line 326, in map_exceptions
    yield
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1492, in _send_single_request
    (status_code, headers, stream, ext) = await transport.arequest(
  File "/usr/local/lib/python3.8/site-packages/httpx/_transports/default.py", line 169, in arequest
    return await self._pool.arequest(
  File "/usr/local/lib/python3.8/site-packages/httpcore/_async/connection_pool.py", line 218, in arequest
    response = await connection.arequest(
  File "/usr/local/lib/python3.8/site-packages/httpcore/_async/connection.py", line 106, in arequest
    return await self.connection.arequest(method, url, headers, stream, ext)
  File "/usr/local/lib/python3.8/site-packages/httpcore/_async/http11.py", line 72, in arequest
    ) = await self._receive_response(timeout)
  File "/usr/local/lib/python3.8/site-packages/httpcore/_async/http11.py", line 133, in _receive_response
    event = await self._receive_event(timeout)
  File "/usr/local/lib/python3.8/site-packages/httpcore/_async/http11.py", line 172, in _receive_event
    data = await self.socket.read(self.READ_NUM_BYTES, timeout)
  File "/usr/local/lib/python3.8/site-packages/httpcore/_backends/asyncio.py", line 157, in read
    raise
  File "/usr/local/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.8/site-packages/httpcore/_exceptions.py", line 12, in map_exceptions
    raise to_exc(exc) from None
httpcore.ReadError: [Errno 104] Connection reset by peer

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/envoy_reader/envoy_reader.py", line 125, in getData
    response = await client.get(
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1539, in get
    return await self.request(
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1361, in request
    response = await self.send(
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1396, in send
    response = await self._send_handling_auth(
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1434, in _send_handling_auth
    response = await self._send_handling_redirects(
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1466, in _send_handling_redirects
    response = await self._send_single_request(request, timeout)
  File "/usr/local/lib/python3.8/site-packages/httpx/_client.py", line 1492, in _send_single_request
    (status_code, headers, stream, ext) = await transport.arequest(
  File "/usr/local/lib/python3.8/contextlib.py", line 131, in __exit__
    self.gen.throw(type, value, traceback)
  File "/usr/local/lib/python3.8/site-packages/httpx/_exceptions.py", line 343, in map_exceptions
    raise mapped_exc(message, **kwargs) from exc  # type: ignore
httpx.ReadError: [Errno 104] Connection reset by peer

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 175, in _async_refresh
    self.data = await self._async_update_data()
  File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 145, in _async_update_data
    return await self.update_method()
  File "/usr/src/homeassistant/homeassistant/components/enphase_envoy/__init__.py", line 52, in async_update_data
    await envoy_reader.getData()
  File "/usr/local/lib/python3.8/site-packages/envoy_reader/envoy_reader.py", line 137, in getData
    response.raise_for_status()
UnboundLocalError: local variable 'response' referenced before assignment
```

Reply:
@bdraco I was testing this against an Envoy running `D5.0.49` and noticed the local variable error, but also noticed that I was not able to retrieve the Inverter data. This envoy has the standard login of envoy for the username and the last 6 digits of the serial number for the password. I tried entering the credentials manually but I got the same response.

I didn't run it in Home Assistant just did a quick test from the command line before moving to the integration with HA.

```
Reading...
production:              local variable 'production' referenced before assignment
consumption:             511
daily_production:        local variable 'daily_production' referenced before assignment
daily_consumption:       1982
seven_days_production:   local variable 'seven_days_production' referenced before assignment
seven_days_consumption:  167894
lifetime_production:     local variable 'lifetime_production' referenced before assignment
lifetime_consumption:    42204657
inverters_production:    Inverter data not available for your Envoy device.
```

Reply:
I moved it before the class. It should work correctly now